From 8624cab29c52db9052bf022683cfd3d11369cc0d Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Tue, 2 Apr 2013 08:17:43 +0200
Subject: [PATCH 103/105] BUILD: add explicit support for Mac OS/X

The "osx" target may now be passed in the TARGET variable. It supports
the same features as FreeBSD and allows its users to use the GNU makefile
instead of the platform-specific makefile which lacks some features.
---
 Makefile | 8 ++++++++
 README   | 1 +
 2 files changed, 9 insertions(+)

diff --git a/Makefile b/Makefile
index da6efe9..b66a07f 100644
--- a/Makefile
+++ b/Makefile
@@ -269,6 +269,13 @@ ifeq ($(TARGET),freebsd)
   USE_TPROXY     = implicit
   USE_LIBCRYPT   = implicit
 else
+ifeq ($(TARGET),osx)
+  # This is for Mac OS/X
+  USE_POLL       = implicit
+  USE_KQUEUE     = implicit
+  USE_TPROXY     = implicit
+  USE_LIBCRYPT   = implicit
+else
 ifeq ($(TARGET),openbsd)
   # This is for OpenBSD >= 3.0
   USE_POLL       = implicit
@@ -291,6 +298,7 @@ ifeq ($(TARGET),cygwin)
 endif # cygwin
 endif # aix52
 endif # openbsd
+endif # osx
 endif # freebsd
 endif # solaris
 endif # linux2628
diff --git a/README b/README
index 0da3720..01cb99d 100644
--- a/README
+++ b/README
@@ -28,6 +28,7 @@ and assign it to the TARGET variable :
   - linux2628   for Linux 2.6.28 and above (enables splice and tproxy)
   - solaris     for Solaris 8 or 10 (others untested)
   - freebsd     for FreeBSD 5 to 8.0 (others untested)
+  - osx         for Mac OS/X
   - openbsd     for OpenBSD 3.1 to 5.2 (others untested)
   - aix52       for AIX 5.2
   - cygwin      for Cygwin
-- 
1.8.1.5

